### PR TITLE
Add ability to restrict static analysis to particular paths

### PR DESCRIPTION
**Pull request by schmerl**

### CODE REVIEW

File: src/rosdiscover/cli.py [hunk 1]
```diff
@@ -145,6 +145,12 @@ def main() -> None:
         nargs='+',
         help='the paths of the translation unit source files for this node, relative to the package directory',
     )
+    p.add_argument(
+        'restrict-to',
+        nargs='+',
+        type=str,
+        help='the paths to restrict static analysis to, relative to the package directory'
```
Comment (ChrisTimperley) [inline src/rosdiscover/cli.py:152]: Ah. These are actually absolute paths.

File: src/rosdiscover/config.py [hunk 2]
```diff
@@ -96,12 +99,19 @@ def from_dict(cls, dict_: t.Mapping[str, t.Any]) -> 'NodeSourceInfo':
                 raise ValueError("expected 'entrypoint' to be a string")
             entrypoint = dict_['entrypoint']
 
+        restricted_paths = []
+        if 'restrict-analysis-to-paths' in dict_:
+            if not isinstance(dict_['restrict-analysis-to-paths'], list):
+                raise ValueError("expected 'restrict-analysis-to-paths' to bae a list")
+            restricted_paths = dict_['restrict-analysis-to-paths']
```
Comment (ChrisTimperley) [inline src/rosdiscover/config.py:106]: very minor suggestion: you can just define `restricted_paths = dict_.get("restrict-analysis-to-paths", [])` after you've done the sanity checking above.

File: src/rosdiscover/recover/tool.py [hunk 3]
```diff
@@ -352,8 +354,11 @@ def _recover(
             shlex.quote(os.path.dirname(compile_commands_path)),
             "-output-filename",
             json_model_filename,
-            ' '.join(shlex.quote(p) for p in source_file_abs_paths),
         ]
+        for path in restrict_to_paths:
+            args = args + ['--restrict-to', path]
```
Comment (ChrisTimperley) [inline src/rosdiscover/recover/tool.py:359]: very minor: `args += [...]`

File: src/rosdiscover/recover/tool.py [hunk 4]
```diff
@@ -352,8 +354,11 @@ def _recover(
...
+
+        args = args + [' '.join(shlex.quote(p) for p in source_file_abs_paths)]
```
Comment (ChrisTimperley) [inline src/rosdiscover/recover/tool.py:361]: same as above

**Review by ChrisTimperley — APPROVED** [review]
One important note: the paths should be absolute (and we should probably check that they are so). Otherwise, this looks good 👍🏻

**Comment by schmerl** [comment]
Addressed all the review comments.

**Pull request merged**